build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/task/DistributeTask.java

```diff
@@ -0,0 +1,75 @@
+package org.jfrog.gradle.plugin.artifactory.task;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.util.EntityUtils;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.TaskAction;
+import org.jfrog.build.api.release.Distribution;
+import org.jfrog.build.extractor.clientConfiguration.ArtifactoryClientConfiguration;
+import org.jfrog.build.extractor.clientConfiguration.client.ArtifactoryBuildInfoClient;
+import org.jfrog.gradle.plugin.artifactory.dsl.ArtifactoryPluginConvention;
+import org.jfrog.gradle.plugin.artifactory.dsl.DistributerConfig;
+import org.jfrog.gradle.plugin.artifactory.extractor.GradleClientLogger;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class DistributeTask extends DefaultTask {
+
+    public static final String DISTRIBUTE_TASK_NAME = "artifactoryDistribute";
+    private static final Logger log = Logging.getLogger(DistributeTask.class);
+
+    @TaskAction
+    public void distributeBuild() throws IOException {
+        validate();
+
+        ArtifactoryPluginConvention convention = (ArtifactoryPluginConvention) getProject().getConvention().getPlugins().get("artifactory");
+        DistributerConfig distributerConfig = convention.getDistributerConfig();
+        ArtifactoryClientConfiguration clientConf = convention.getClientConfig();
+
+        ArtifactoryBuildInfoClient infoClient = new ArtifactoryBuildInfoClient(
```

Comment:
ArtifactoryBuildInfoClient should be closed:
```suggestion
        try(ArtifactoryBuildInfoClient infoClient = new ArtifactoryBuildInfoClient(...
```

Reply:
Good catch! Thanks!